logicrunner/builtin/contract/rootdomain/rootdomain.go

```diff
@@ -85,31 +87,43 @@ func (rd *RootDomain) GetBurnAddress() (string, error) {
 }
 
 // GetMemberByPublicKey gets member reference by public key.
-func (rd RootDomain) GetMemberByPublicKey(publicKey string) (insolar.Reference, error) {
-	result, ok := rd.PublicKeyMap[trimPublicKey(publicKey)]
-	if !ok {
-		return insolar.Reference{}, fmt.Errorf("member for this public key does not exist")
+func (rd RootDomain) GetMemberByPublicKey(publicKey string) (*insolar.Reference, error) {
+	trimmedPublicKey := trimPublicKey(publicKey)
+	i := foundation.GetShardIndex(trimmedPublicKey, insolar.GenesisAmountPublicKeyShards)
+	if i >= len(rd.PublicKeyShards) {
+		return nil, fmt.Errorf("incorect shard index")
 	}
-	ref, err := insolar.NewReferenceFromBase58(result)
+	s := shard.GetObject(rd.PublicKeyShards[i])
+	refStr, err := s.GetRef(trimmedPublicKey)
 	if err != nil {
-		return insolar.Reference{}, errors.Wrap(err, "bad member reference for this public key")
+		return nil, errors.Wrap(err, "failed to get reference in shard")
+	}
+	ref, err := insolar.NewReferenceFromBase58(refStr)
+	if err != nil {
+		return nil, errors.Wrap(err, "bad member reference for this public key")
 	}
 
-	return *ref, nil
+	return ref, nil
 }
 
-// GetMemberByBurnAddress gets member reference by burn address.
-func (rd RootDomain) GetMemberByBurnAddress(burnAddress string) (insolar.Reference, error) {
-	result, ok := rd.BurnAddressMap[trimBurnAddress(burnAddress)]
-	if !ok {
-		return insolar.Reference{}, fmt.Errorf("member for this migration address does not exist")
+// GetMemberByMigrationAddress gets member reference by burn address.
+func (rd RootDomain) GetMemberByMigrationAddress(migrationAddress string) (*insolar.Reference, error) {
```

Comment:
why by pointer insolar.Reference?
make (rd RootDomain) at pointer

Reply:
why do we use root domain fo this case?

Reply:
It returns pointer because it is more beautiful to return nil except insolar.Reference{}. 
why make rd pointer? this method doesn't change anything